core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -95,8 +96,21 @@
   private final String userAgent;
   private final Object lock = new Object();
 
+  /* Compression related */
+  /**
+   *  When a client connects to a server, it does not know what encodings are supported.  This set
+   *  is the union of all accept-encoding headers the server has sent.  It is used to pick an
+   *  encoding when contacting the server again.  One problem with the gRPC protocol is that if
+   *  there is only one RPC made (perhaps streaming, or otherwise long lived) an encoding will not
+   *  be selected.  To combat this you can preflight a request to the server to fill in the mapping
+   *  for the next one.  A better solution is if you have prior knowledge that the server supports
+   *  an encoding, and fill this structure before the request.
```

Comment:
Yes, but I didn't want to commit too early to such a design.  Also, in the spirit of keeping PRs small, I left it out.